### PR TITLE
Update test instructions and remove 2.2

### LINKED ISSUE #27923: Unclear docs on test from cmd line
On https://github.com/dotnet/efcore/blob/main/docs/getting-and-building-the-code.md

It is not clear if test.cmd builds and runs test (as per the table) or requires build first:

sentence further on: Tests can be run on the command line (after build)

Happy to do a PR - assume the table must be updated.

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #27923

### CODE REVIEW

File: docs/getting-and-building-the-code.md [hunk 1]
```diff
@@ -81,79 +81,6 @@ test
 2. Clean the source directory. `git clean -xid` will clean files in the EF source directory.
 3. Clear nuget packages and caches. `nuget.exe locals all -clear` will delete the NuGet caches. (You can get nuget.exe from <https://dist.nuget.org/index.html> or use `dotnet nuget`).
 
-***
```
Comment (ajcvickers) [inline docs/getting-and-building-the-code.md:84]: @dotnet/efteam Do we want to remove 2.x instructions? It's not supported, but people may still want to build it for unsupported use? But, on the other hand, are we encouraging this by keeping these instructions here?

Comment (ErikEJ) [reply docs/getting-and-building-the-code.md:84]: They will remain in older versions of the file..

Comment (roji) [reply docs/getting-and-building-the-code.md:84]: I vote to kill

**Review by AndriySvyryd — APPROVED** [review]

**Comment by AndriySvyryd** [comment]
Thanks!

**Pull request merged**